apiserver/facades/agent/uniter/uniter.go

```diff
@@ -2598,18 +2602,32 @@ func (u *UniterAPI) goalStateRelations(allRelations []*state.Relation) (map[stri
 			} else {
 				return nil, err
 			}
-			goalState := params.GoalStateStatus{}
-			statusInfo, err := r.Status()
-			if err != nil {
-				return nil, errors.Annotate(err, "getting relation status")
+
+			// Peer units are shown in the Units section, not here.
```

Comment:
Why doesn't the `e.Relation.Role == "peer"` check above cover this case? Is it because the relation might be a normal provider/requirer one but to the same application? Maybe worth expanding the comment?

Reply:
The comment was poorly worded. I've updated to make it clearer